.travis.yml

```diff
@@ -1,8 +1,32 @@
 language: node_js
 node_js:
-  - "4.2"
-  - "stable"
+- "4.2"
+- "stable"
 sudo: false
 cache:
   directories:
   - node_modules
+after_script:
+- |
+  # RELEASE_BRANCHES and NPM_TOKEN defined in Travis settings panel
+  declare exitCode
+  $(npm bin)/travis-after-all
+  exitCode=$?
+  if [[
+    # Execute after all jobs finish successfully
+    $exitCode = 0 &&
+    # Only release on release branches
+    $RELEASE_BRANCHES =~ $TRAVIS_BRANCH &&
+    # Don't release on PR builds
+    $TRAVIS_PULL_REQUEST = "false"
+  ]]; then
+    # Authenticate NPM
+    echo "//registry.npmjs.org/:_authToken=\${NPM_TOKEN}" > .npmrc
+    # Set version to commit SHA
+    npm --no-git-tag-version version $(node -p -e "require('./package.json').version")-${TRAVIS_COMMIT:0:5}
```

Comment:
I tried using the `0.1.0-[epoch]` style but it didn't seem to work with a package requiring `"scratch-render": "^0.1.0"` (npm complained that the version couldn't be found). However, I found that using `0.1.0-[release label].[increment]` does work, so I went with that.